### PR TITLE
Create directory for monitoring data

### LINKED ISSUE #1549: [BUG] Monitoring data is not preserved b/w system upgrades
**Describe the bug**

Metrics collected by our monitoring app are reset after each system upgrade. This is true for both the old InfluxDB-based stack and the new Prometheus-based one:

* InfluxDB keeps data in [/var/lib/data](https://github.com/gravitational/monitoring-app/blob/version/5.5.x/resources/influxdb.yaml#L101) inside planet which is not mounted from host.
* Prometheus currently uses emptyDir volume for its data.

For container metrics this is not a huge issue since they are recreated during upgrade and their metrics get reset anyway, but this also makes it impossible to look at historic data for node metrics.

**To Reproduce**

Perform an upgrade where a system upgrade is involved and look at the Monitoring tab.

**Expected behavior**

- [ ] InfluxDB data persists b/w system upgrades.
- [ ] Prometheus data persists b/w system upgrades and pod restarts.

Bonus point:

- [ ] Is there an easy way to migrate data during an upgrade? For Prometheus we're probably out of luck since it's in the pod's ephemeral storage, but InfluxDB keeps it on disk.

**Environment (please complete the following information):**
 - Gravity [e.g. 5.5.4]: all

### PR DESCRIPTION
**Pull request by UnderGreen**
## Description
 - Create directory for monitoring data to make them persistent during planet upgrade.
 - Use service-uid/gid from config instead of defaults while set permissions for directories.

## Type of change

* Bug fix (non-breaking change which fixes an issue)

## Linked tickets and other PRs

Requires migration implementation in monitoring application, which is in progress.

* Closes #1549 for pre-Promethues branch

## Testing done

- [ ] Self-review the change
- [ ] Write tests
- [X] Perform manual testing
- [ ] Write documentation
- [] Address review feedback

### CODE REVIEW

File: lib/install/phases/bootstrap.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"path/filepath"
 	"strconv"
 
+	"github.com/davecgh/go-spew/spew"
```
Comment (a-palchikov) [inline lib/install/phases/bootstrap.go:27]: nit: move to next import group

File: lib/install/phases/bootstrap.go [hunk 2]
```diff
@@ -203,6 +204,11 @@ func (p *bootstrapExecutor) configureSystemDirectories() error {
 		filepath.Join(stateDir, "secrets"),
 		filepath.Join(stateDir, "backup"),
 	}
+	p.Info(spew.Sdump(p.Application))
```
Comment (a-palchikov) [inline lib/install/phases/bootstrap.go:207]: Debugging aid or necessary?

File: lib/install/phases/bootstrap.go [hunk 3]
```diff
@@ -203,6 +204,11 @@ func (p *bootstrapExecutor) configureSystemDirectories() error {
...
+	if p.Application.Manifest.Extensions != nil && p.Application.Manifest.Extensions.Monitoring != nil && !p.Application.Manifest.Extensions.Monitoring.Disabled {
```
Comment (a-palchikov) [inline lib/install/phases/bootstrap.go:208]: Does it need to be conditionally enabled? What happens if we always create/chown this directory?

Comment (UnderGreen) [reply lib/install/phases/bootstrap.go]: Just you know, to make things not hardcoded. Users could ask why do I have empty monitoring directories on all servers even if I disabled monitoring extension.

File: lib/install/phases/bootstrap.go (hunk 2 above)
Comment (UnderGreen) [reply lib/install/phases/bootstrap.go]: PR was an in draft stage :).
Now it is open for review, can you reviet it again? @a-palchikov

File: lib/install/phases/bootstrap.go [hunk 4]
```diff
@@ -363,6 +371,13 @@ func (p *bootstrapExecutor) configureServiceUser() error {
 	return nil
 }
 
+func (p *bootstrapExecutor) isMonitoringEnabled() bool {
+	if p.Application.Manifest.Extensions != nil && p.Application.Manifest.Extensions.Monitoring != nil && p.Application.Manifest.Extensions.Monitoring.Disabled {
```
Comment (r0mant) [inline lib/install/phases/bootstrap.go:375]: To be honest, I had the same question as @a-palchikov. Why not simplify things and just always create it? /var/lib/gravity is our state directory and we control what we put in it, users should not be concerned with what directories we need there.

**Review by r0mant — COMMENTED** [review]
Also, what about upgrades? IIRC during upgrade directories created in a different place.

File: lib/install/phases/bootstrap.go (hunk 4 above)
Comment (UnderGreen) [reply lib/install/phases/bootstrap.go:371]: Ok, changed it to be always created.

File: lib/install/phases/bootstrap.go [hunk 5]
```diff
@@ -202,7 +202,9 @@ func (p *bootstrapExecutor) configureSystemDirectories() error {
 		filepath.Join(stateDir, "site", "packages", "tmp"),
 		filepath.Join(stateDir, "secrets"),
 		filepath.Join(stateDir, "backup"),
+		filepath.Join(stateDir, "monitoring"),
```
Comment (r0mant) [inline lib/install/phases/bootstrap.go:205]: I think we also need to add it here: https://github.com/gravitational/gravity/blob/version/5.5.x/lib/ops/opsservice/deploy.go#L88. This is where directories get created/updated during an upgrade.

**Review by r0mant — APPROVED** [review]

**Pull request merged**